src/mscorlib/src/System/Random.cs

```diff
@@ -114,11 +121,39 @@ public Random()
           return retVal;
       }
 
+
+      [ThreadStatic]
+      private static Random t_threadRandom;
+      private static Random s_globalRandom = new Random(GenerateGlobalSeed());
+
+      /*=====================================GenerateSeed=====================================
+      **Returns: An integer that can be used as seed values for consecutively
+                 creating lots of instances on the same thread within a short period of time.
+      ========================================================================================*/
+      private static int GenerateSeed() {
+          if (t_threadRandom == null) {
+              int seed;
+              lock (s_globalRandom) {
+                  seed = s_globalRandom.Next();
+              }
+              t_threadRandom = new Random(seed);
+          }
+          return t_threadRandom.Next();
+      }
+
+      /*==================================GenerateGlobalSeed====================================
+      **Action:  Creates a number to use as global seed.
+      **Returns: An integer that is safe to use as seed values for thread-local seed generators.
+      ==========================================================================================*/
+      private static int GenerateGlobalSeed() {
+          return Guid.NewGuid().GetHashCode();
```

Comment:
And for fun I also tried `new RNGCryptoServiceProvider().GetBytes(new byte[4]);`:

```
32467
33
27
27
21
18
21
21
```

So that is indeed more expensive that the Guid solution. 3.2 milliseconds.